cache.go

```diff
@@ -119,6 +121,14 @@ func (c *Cache) Clear() {
 // is called are likely to panic
 func (c *Cache) Stop() {
 	close(c.promotables)
+	<-c.donec
+}
+
+func (c *Cache) restart() {
```

Comment:
What do you think of having `New` call restart so that these 4 lines aren't repeated?

```
c := &Cache{
 		list:          list.New(),
 		Configuration: config,
 		bucketMask:    uint32(config.buckets) - 1,
 		buckets:       make([]*bucket, config.buckets)
 	}

for i := 0 .... {
}
c.restart()
return c
```

??